internal/pkg/utils/winpol/windows_policy.go

```diff
@@ -85,13 +101,64 @@ func CalculateEndpointPolicies(
 		encoded, err := json.Marshal(dict)
 		if err != nil {
 			logger.WithError(err).Error("Failed to add outbound NAT exclusion.")
-			return nil, err
+			return nil, nil, err
 		}
 
-		outputPols = append(outputPols, json.RawMessage(encoded))
+		outputV1Pols = append(outputV1Pols, json.RawMessage(encoded))
+
+		// Convert v2 policy. If the conversion to V2 policy fails just log and
+		// continue.
+		v2Pol, err := convertToHcnEndpointPolicy(dict)
+		if err != nil {
+			logger.WithError(err).Warnf("Failed to convert endpoint policy to HCN endpoint policy: %+v", dict)
+		} else {
+			outputV2Pols = append(outputV2Pols, v2Pol)
+		}
 	}
 
-	return outputPols, nil
+	return outputV1Pols, outputV2Pols, nil
+}
+
+// convertToHcnEndpointPolicy converts a map representing the raw data of a V1
+// policy and converts it to an HCN endpoint policy.
+//
+// For example, we convert from raw JSON like:
+//
+// {
+//   "Type":  "OutBoundNAT",
+//   "ExceptionList":  [
+//     "10.96.0.0/12",
+//     "192.168.0.0/16"
+//   ]
+// }
+//
+// to:
+//
+// hcn.EndpointPolicy{
+//   Type: hcn.OutBoundNAT,
+//   Settings: json.RawMessage(
+//     []byte(`{"ExceptionList":["10.96.0.0/12","192.168.0.0/16"]}`),
+//   ),
+// }
+func convertToHcnEndpointPolicy(policy map[string]interface{}) (hcn.EndpointPolicy, error) {
+	hcnPolicy := hcn.EndpointPolicy{}
+
+	// Get v2 policy type.
+	policyType, ok := policy["Type"].(hcn.EndpointPolicyType)
```

Comment:
Is this type assertion correct? The type is `string` so it always return false?

Reply:
I think the best is to remove the check all together. We are not validate the type for v1 policy either. It is up to the user to make sure the policy is correct. Besides, cni would log error (invalid parameter?) anyway if it applies an invalid policy.

Reply:
> Is this type assertion correct? The type is string so it always return false?

Yeah not sure what I was thinking 😅 Anyway you fixed this up for me